### PR TITLE
Mocha testing

### PR DESCRIPTION
**Pull request by mayppong**
Here's a sample of how to write simple test cases for sockets. I've written up 2: one for setting team name and one for setting player's initials.

To test it out, run
-  `npm install` to install the test dependencies
- `npm start` to start the web server. We have to do this manually for now
- `./node_modules/.bin/mocha` to run the tests

### CODE REVIEW

File: lib/player.js [hunk 1]
```diff
@@ -45,12 +56,13 @@ var Player = {
 	},
 
 	// save player initials
-	update_initials: function(socket) {
+	update_initials: function(socket, initials) {
 		if (!socket || !socket.player_id || !socket.initials){
```
Comment (whiterook6) [inline lib/player.js]: if you're taking in the initials, no need to check them here, or to have them set on the socket in the first place.

File: lib/team.js [hunk 2]
```diff
@@ -1,4 +1,44 @@
 var Team = {
+	enable: function(socket) {
+		socket.on('set_team', function(data){
+			if (socket.team){
```
Comment (whiterook6) [inline lib/team.js:8]:
`````` javascript
if (socket.team && data == socket.team){
  return;
}```
``````

File: package.json [hunk 3]
```diff
@@ -13,5 +13,9 @@
 		"sequelize": "~3.9.0",
 		"sequelize-cli": "~1.9.1"
 	},
+	"devDependencies": {
```
Comment (whiterook6) [inline package.json:16]: is this a special key for packages? Or a custom key?

File: test/team_test.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+		'force new connection': true
+	}
+};
+
+/**
+ * Tests
+ */
+describe('Team Module', function() {
+	describe('set team', function() {
+		it('should response with new team name', function(done) {
```
Comment (whiterook6) [inline test/team_test.js]: should respond with a new team name

File: lib/player.js [hunk 5]
```diff
@@ -1,4 +1,15 @@
 var Player = {
+	enable: function(socket) {
+		this.create_player(socket);
+
+		socket.on('set_initials', function(data){
+			if (data && data != socket.initials){
+				Player.update_initials(socket, data);
+				Player.reset_score(socket);
+			}
+			io.sockets.emit('user.renamed', socket.initials);
```
Comment (whiterook6) [inline lib/player.js:10]: Is this here just to support testing?

Comment (mayppong) [inline lib/player.js:10]: Yes and no. It's good for testing but I think it's also nice if we want the server to be able to set data as well. It's a future proof thing.

File: package.json (hunk 3 above)
Comment (mayppong) [inline package.json:16]: The packages in devDependencies won't get install if you run `npm` in production environment.

**Pull request merged**